stonesoup/metricgenerator/manager.py

```diff
@@ -74,30 +67,65 @@ def generate_metrics(self):
             Metrics generated
         """
 
-        if self.associator is not None and self.association_set is None:
-            self.associate_tracks()
-
         metrics = {}
-        for generator in self.generators:
+
+        generators = self.generators if isinstance(self.generators, list) else [self.generators]
+
+        for generator in generators:
+            if isinstance(generator, SIAPMetrics):
+                if self.associator is not None:
+                    self.associate_tracks(generator)
             metric_list = generator.compute_metric(self)
             # If not already a list, force it to be one below
             if not isinstance(metric_list, list):
                 metric_list = [metric_list]
             for metric in metric_list:
-                metrics[metric.title] = metric
-        return metrics
+                if generator.generator_name not in metrics.keys():
+                    metrics[generator.generator_name] = {metric.title: metric}
+                else:
+                    metrics[generator.generator_name][metric.title] = metric
+
+        self.metrics = metrics
 
-    def list_timestamps(self):
+        return self.metrics
+
+    def list_timestamps(self, generator):
         """List all the timestamps used in the tracks and truth, in order
 
         Returns
         ----------
         : list of :class:`datetime.datetime`
             unique timestamps present in the internal tracks and truths.
         """
-
         # Make a list of all the unique timestamps used
         timestamps = {state.timestamp
-                      for sequence in chain(self.tracks, self.groundtruth_paths)
+                      for sequence in chain(self.states_sets[generator.tracks_key],
+                                            self.states_sets[generator.truths_key])
                       for state in sequence}
+
         return sorted(timestamps)
+
+    def display_basic_metrics(self):
+        """
+        Print basic metrics generated for each :class:`BasicMetrics` generator.
+        """
+        for generator_name, generator in self.metrics.items():
+            if "Number of targets" in generator.keys():
+                print(f'\nGenerator: {generator_name}')
+                for metric_key, metric in generator.items():
+                    if isinstance(metric.generator, BasicMetrics):
+                        print(f"{metric.title}: {metric.value}")
+
+    def get_siap_averages(self, generator_name):
+        """
+        Get SIAP averages metrics from SIAP metric generator specified by generator_name
+
+        Returns
+        ----------
+        : dict of :class`SIAPMetrics` averages
+        """
+        siap_metrics = self.metrics[generator_name]
+        siap_averages = {siap_metrics.get(metric) for metric in siap_metrics
+                         if metric.startswith("SIAP") and not metric.endswith(" at times")}
+
+        return siap_averages
```

Comment:
These methods also seem specific to the name or specific metric. A different approach is needed here.